numpy/core/src/multiarray/convert_datatype.c

```diff
@@ -1927,24 +1928,34 @@ PyArray_Zero(PyArrayObject *arr)
         return NULL;
     }
 
-    obj=PyInt_FromLong((long) 0);
+    if (global_zero == NULL)
+    {
+        global_zero = PyInt_FromLong((long) 0);
+        if (global_zero == NULL)
+        {
+            PyErr_SetNone(PyExc_MemoryError);
```

Comment:
the python function should already set the correct exception, so no special handling is necessary
Also our if brace style is:

```
if (cond) {
}
```